src/Elmish.WPF/ViewModel.fs

```diff
@@ -684,6 +695,11 @@ and [<AllowNullLiteral>] internal ViewModel<'model, 'msg>
           |> ValueOption.map (fun vm -> b.SubModelSeqBinding.GetId vm.CurrentModel)
         b.Set id model
         true
+    | Cached b ->
+        let successful = trySetMember model value b.Binding
+        if successful then
+          b.Cache := None // `None` is safter, but consider using `Some value` instead
```

Comment:
This can be moved up after `then` since it's so short (but I won't fight you if you prefer it like this)

Reply:
I will keep it like this because it is a mutation.